rir/src/compiler/opt/inline.cpp

```diff
@@ -211,18 +211,26 @@ class TheInliner {
                         if (ld) {
                             Value* a = arguments[ld->id];
                             if (auto mk = MkArg::Cast(a)) {
-                                // We need to cast from a promise to a lazy
-                                // value
-                                auto cast =
-                                    new CastType(a, RType::prom,
-                                                 mk->isEager()
-                                                     ? mk->eagerArg()
-                                                           ->type.forced()
-                                                           .promiseWrappedVal()
-                                                     : ld->type);
-                                ip = bb->insert(ip + 1, cast);
-                                ip--;
-                                a = cast;
+                                if (!ld->type.maybePromiseWrapped()) {
```

Comment:
no, but it's a bit faster. In the case the promise is eager, and the callee was already compiled under the assumption that it will receive an already evaluated promise.

Instead of:
```
// caller
val  e = ldconst 123
prom p = mkpromise(e, ...)
//callee
val^ l = castType(p, lazy)
val  v = force(l)
         use(v)
```

we directly generate this

```
val  e = ldconst 123
prom p = mkpromise(e, ...)
val  v = force(e)
         use(v)
```

both of those will end up being converted to 

```
val  e = ldconst 123
         use(e)
```

just the fromer should avoid some extra steps